packages/mini-oxygen/src/vite/utils.ts

```diff
@@ -10,13 +10,12 @@ import type {ViteDevServer} from 'vite';
  */
 export function toURL(req: string | IncomingMessage = '/', origin?: string) {
   const isRequest = typeof req !== 'string';
-  const pathname = (isRequest ? req.url : req) || '/';
```

Comment:
Instead of using the second argument to `new URL`, just manually concatenate the origin with the pathname.